src/components/ResultScreen/index.tsx

```diff
@@ -0,0 +1,273 @@
+  word: string
+  translation: string
+}
+export type ResultSpeedInfo = {
+  speed: string
+  minute: number
+  second: number
+}
+
+type ResultScreenProps = {
+  incorrectInfo: IncorrectInfo[]
+  speedInfo: ResultSpeedInfo
+  repeatButtonHandler: () => void
+  invisibleButtonHandler: () => void
+  nextButtonHandler: () => void
+}
+
+export const ResultScreen: React.FC<ResultScreenProps> = ({
+  speedInfo,
+  incorrectInfo,
+  repeatButtonHandler,
+  invisibleButtonHandler,
+  nextButtonHandler,
+}) => {
+  const wordList = useWordList()
+
+  const isLastChapter = useMemo(() => {
+    if (wordList) {
+      return wordList.chapter === wordList.chapterListLength - 1
+    } else {
+      return false
+    }
+  }, [wordList])
+
+  const isDarkMode = useMemo(() => {
+    return document.documentElement.classList.contains('dark')
+  }, [])
+
+  const correctRate = useMemo(() => {
+    const chapterLength = wordList?.words.length || 0
+    const correctCount = chapterLength - incorrectInfo.length
+    return Math.floor((correctCount / chapterLength) * 100)
+  }, [wordList, incorrectInfo])
+
+  const rootFontSize = useMemo(() => {
+    const root = document.documentElement
+    return parseInt(window.getComputedStyle(root).getPropertyValue('font-size'))
+  }, [])
+
+  const mistakeLevel = useMemo(() => {
+    if (correctRate >= 85) {
+      return 0
+    } else if (correctRate >= 70) {
+      return 1
+    } else {
+      return 2
+    }
+  }, [correctRate])
+
+  const timeString = useMemo(() => {
+    const { minute, second } = speedInfo
+    const minuteString = minute < 10 ? '0' + minute : minute + ''
+    const secondString = second < 10 ? '0' + second : second + ''
+    return `${minuteString}:${secondString}`
+  }, [speedInfo])
+
+  const conclusion = () => {
+    let content
+    switch (mistakeLevel) {
+      case 0:
+        content = (
+          <>
+            <FontAwesomeIcon icon={['fas', 'heart']} className="text-indigo-600 pt-2" size="lg" />
+            <span className="text-base font-medium text-gray-700 ml-2 leading-10 inline-block align-middle">
+              表现不错！只错了 {incorrectInfo.length} 个单词
+            </span>
+          </>
+        )
+        break
+      case 1:
+        content = (
+          <div className="">
+            <FontAwesomeIcon icon={['fas', 'thumbs-up']} className="text-indigo-600 leading-10 " size="lg" />
+            <div className="text-base font-medium text-gray-700 ml-2 leading-10 inline-block align-middle">
+              有些小问题哦，下一次可以做得更好！
+            </div>
+          </div>
+        )
+        break
+      case 2:
+        content = (
+          <>
+            <FontAwesomeIcon icon={['fas', 'exclamation-triangle']} className="text-indigo-600 pt-2" size="lg" />
+            <div className="text-base font-medium text-gray-700 ml-2 leading-10 inline-block align-middle">错误太多，再来一次如何？</div>
+          </>
+        )
+        break
+    }
+    return <div className="h-10">{content}</div>
+  }
```

Comment:
Why not extract this part to a new component? Let the `mistakeLevel` be the prop of the new component.

Reply:
OK I'll extract Conclusion out.